src/modules/menu.cpp

```diff
@@ -74,38 +74,29 @@ namespace modules {
       builder->cmd_close();
     } else if (tag == TAG_MENU && m_level > -1) {
       auto spacing = m_formatter->get(get_format())->spacing;
+      //Insert spacing after menu-items and before menu-toggle for expand-right=true
+      if (m_expand_right) {
+        if (*m_labelseparator)
+          builder->node(m_labelseparator);
+        builder->space(spacing);
```

Comment:
If I am not wrong this space should only be printed if there is a separator.
As you can see in the following screenshot, it seems that there is two spaces between the "x" and "Browsers"
![poly](https://user-images.githubusercontent.com/10365269/57944243-61a9d300-78a4-11e9-874d-4fd0490c9a49.png)

Reply:
That's correct: The spacing should have only been printed if the user specified a separator. I've made a commit to address this. I didn't test this code without a separator, originally. I've tested this new commit thoroughly with with a monospace font and all variations of `expand-right` and `label-separator`.

![Screenshot_2019-05-17_17-59-45](https://user-images.githubusercontent.com/5731041/57960413-32a85700-78ce-11e9-9571-d7dd57d3a7ff.png)
![Screenshot_2019-05-17_17-58-59](https://user-images.githubusercontent.com/5731041/57960415-36d47480-78ce-11e9-9d65-b7f89a0240c2.png)


The correct `format-spacing` in the `menu` module is already printed between the menu-toggle and menu-items. I'm actually unsure where it comes from, which is why this separator+spacing issue came up. I didn't account for it. I don't see any signs of the `menu` module inserting the spacing, so I suspect that the 'builder' class or some formatter or something is magically inserting the right amount of spacing, making my life easier.

Reply:
It's not the builder nor the a formatter, the space is inserted here ;)
https://github.com/polybar/polybar/blob/207d148bfb3a88e7979703929f9f8a4532a2c7f9/src/modules/menu.cpp#L87